### PR TITLE
modified access to ingester shell

### PR DESCRIPTION
**Pull request by anandkp92**
1. restricted access to shell to only localhost
2. included an option to set a password to access the shell in ingester.yml

### CODE REVIEW

File: ingester/config.go [hunk 1]
```diff
@@ -50,6 +50,11 @@ type Config struct {
 	Store struct {
 		Path string
 	}
+
+	IngesterShell struct {
+		PasswordLogin	bool
+		Password 		string
```
Comment (gtfierro) [inline ingester/config.go]: Can you fix the formatting here?

File: ingester/config.go [hunk 2]
```diff
@@ -80,6 +85,9 @@ func init() {
 
 	viper.SetDefault("Store.Path", "store.sqlite3")
 
+	viper.SetDefault("IngesterShell.PasswordLogin", false)
+	viper.SetDefault("IngesterShell.Password", "XBOSWAVEthefuture!")
```
Comment (gtfierro) [inline ingester/config.go]: There should be no defaults for passwords

File: ingester/shell.go [hunk 3]
```diff
@@ -258,7 +258,13 @@ func (ingest *Ingester) shell() {
 		// teardown
 		shell.Close()
 	})
-	if err := ssh.ListenAndServe(":2222", nil, ssh.HostKeyFile("sshkey")); err != nil {
-		logrus.Error(err)
+	if cfg.IngesterShell.PasswordLogin {
+		if err := ssh.ListenAndServe("localhost:2222", nil, ssh.HostKeyFile("sshkey"), ssh.PasswordAuth(func(ctx ssh.Context, pass string) bool { return pass == cfg.IngesterShell.Password })); err != nil {
```
Comment (gtfierro) [inline ingester/shell.go]: The SSH host should be configurable as well

File: ingester/shell.go [hunk 4]
```diff
@@ -258,7 +258,13 @@ func (ingest *Ingester) shell() {
...
+			logrus.Error(err)
+		}
+	} else {
+		if err := ssh.ListenAndServe("localhost:2222", nil, ssh.HostKeyFile("sshkey")); err != nil {
```
Comment (gtfierro) [inline ingester/shell.go]: This should be one `else if` statement

File: README.md [hunk 5]
```diff
@@ -5,6 +5,19 @@
 
 ## Ingester
 
+Requirements:
+1. go
+    ```
+    wget https://dl.google.com/go/go1.11.5.linux-amd64.tar.gz
+    sudo tar -C /usr/local/ -xzf go1.11.5.linux-amd64.tar.gz
+    export PATH=$PATH:/usr/local/go/bin
+    source $HOME/.profile
```
Comment (gtfierro) [inline README.md]: I think we should link to the [official installation instructions](https://golang.org/doc/install). The exporting of the `PATH` variable here doesn't actually affect the `$HOME/.profile` file anyway. You would actually need to add the line to the `.profile` file

File: README.md [hunk 6]
```diff
@@ -25,5 +38,6 @@ How do get started:
 5. Build and run ingester:
     ```
     cd ingester
+    make build
```
Comment (gtfierro) [inline README.md]: This line is unnecessary because the `run` rule already builds the ingester binary

File: ingester/shell.go (hunk 3 above)
Comment (anandkp92) [reply ingester/shell.go]: Do these behaviors look ok:
1) Default for PasswordLogin = False
2) If PasswordLogin == False (or not set), but Password is set, change PasswordLogin = True
3) If PasswordLogin == True, but Password is not set, exit with Error message "Missing Configuration: IngesterShell.Password"
4) Default for SshHostKey = "sshkey"
5) If SshHostKey configured is not a private key file or if it is not present, do not start shell (exit with error message)
Let me know if I need to change any of these.

Comment (gtfierro) [reply ingester/shell.go]: Allow case 2, otherwise those look good

**Pull request merged**